opensearch-operator/api/v1/opensearch_types.go

```diff
@@ -62,22 +62,54 @@ type DashboardsConfig struct {
 	Enable bool `json:"enable,omitempty"`
 }
 
+// Security defines options for managing the opensearch-security plugin
+type Security struct {
+	Tls *TlsConfig `json:"tls,omitempty"`
+	// TBD: securityconfig
+}
+
+// Configure tls usage for transport and http interface
+type TlsConfig struct {
+	Transport *TlsInterfaceConfig `json:"transport,omitempty"`
+	Http      *TlsInterfaceConfig `json:"http,omitempty"`
+	NodesDn   []string            `json:"nodesDn,omitempty"`
+}
+
+// Configure tls usage for the interface
+type TlsInterfaceConfig struct {
+	// If set to true the operator will generate a CA and certificates for the cluster to use, if false secrets with existing certificates must be supplied
+	Generate bool `json:"generate,omitempty"`
+	// Optional, secret that contains the ca certificate
+	CaSecret *TlsSecret `json:"caSecret,omitempty"`
+	// Optional, secret that contains the private key
+	KeySecret *TlsSecret `json:"keySecret,omitempty"`
+	// Optional, secret that contains the certificate for the private key, must be signed by the provided CA
+	CertSecret *TlsSecret `json:"certSecret,omitempty"`
+}
+
+// Reference to a secret
+type TlsSecret struct {
+	SecretName string  `json:"secretName"`
+	Key        *string `json:"key,omitempty"`
+}
+
```

Comment:
The built in [SecretKeySelector](https://pkg.go.dev/k8s.io/api@v0.22.3/core/v1?utm_source=gopls#SecretKeySelector) might be more reusable here

Reply:
I thought about it but SecretKeySelector has `Key` as a required field and I'd like to keep it optional